diff --git a/plugins/paintops/libpaintop/KisSensorPackInterface.h b/plugins/paintops/libpaintop/KisSensorPackInterface.h
index b279854..6bac379 100644
--- a/plugins/paintops/libpaintop/KisSensorPackInterface.h
+++ b/plugins/paintops/libpaintop/KisSensorPackInterface.h
@@ -9,6 +9,7 @@
 
 #include "kritapaintop_export.h"
 #include <QSharedData>
+#include <vector>
 
 struct KisSensorData;
 struct KisCurveOptionDataCommon;
@@ -36,4 +37,4 @@ inline KisSensorPackInterface* QSharedDataPointer<KisSensorPackInterface>::clone
     return d->clone();
 }
 
-#endif // KISSENSORPACKINTERFACE_H
\ No newline at end of file
+#endif // KISSENSORPACKINTERFACE_H
