diff --git a/toolkit/moz.configure b/toolkit/moz.configure
index 43525a0..addbd4d 100644
--- a/toolkit/moz.configure
+++ b/toolkit/moz.configure
@@ -809,19 +809,8 @@ set_config("MOZ_AV1", av1)
 set_define("MOZ_AV1", av1)
 set_config("MOZ_SYSTEM_AV1", depends_if(system_av1)(lambda _: True))
 
-# JXL Image Codec Support
-# ==============================================================
-option("--disable-jxl", help="Disable jxl image support")
-
-
-@depends("--disable-jxl", milestone.is_nightly)
-def jxl(value, is_nightly):
-    if is_nightly and value:
-        return True
-
-
-set_config("MOZ_JXL", jxl)
-set_define("MOZ_JXL", jxl)
+set_config("MOZ_JXL", True)
+set_define("MOZ_JXL", True)
 
 set_config("MOZ_SAMPLE_TYPE_FLOAT32", True)
 set_define("MOZ_SAMPLE_TYPE_FLOAT32", True)
