diff --git a/src/FileFinder.cpp b/src/FileFinder.cpp
index 9ba1443..15e87c0 100644
--- a/src/FileFinder.cpp
+++ b/src/FileFinder.cpp
@@ -19,6 +19,12 @@
 *************************************************************************/
 
 #include <config.h>
+
+#include <cstdlib>
+#include <fstream>
+#include <map>
+#include <set>
+
 #ifdef MIKTEX
 	#include "MiKTeXCom.hpp"
 #else
@@ -31,10 +37,6 @@
 	#endif
 #endif
 
-#include <cstdlib>
-#include <fstream>
-#include <map>
-#include <set>
 #include "FileFinder.hpp"
 #include "FilePath.hpp"
 #include "FileSystem.hpp"
