diff --git a/libffabi/src/ffabi.c b/libffabi/src/ffabi.c
index 166468c..1adb349 100644
--- a/libffabi/src/ffabi.c
+++ b/libffabi/src/ffabi.c
@@ -23,6 +23,7 @@
 #include "mpegaudiodecheader.h"
 #include <libavutil/samplefmt.h>
 #include <libavutil/opt.h>
+#include <libavutil/defs.h>
 #include <string.h>
 #include "internal.h"
 
@@ -374,7 +375,7 @@ FFM_AudioEncodeContext* __cdecl ffm_audio_encode_init(void* logctx,const char* n
 #endif
 
     ctx->avctx->profile = (info->profile != FFM_PROFILE_UNKNOWN) ?
-        info->profile : FF_PROFILE_UNKNOWN;
+        info->profile : AV_PROFILE_UNKNOWN;
 
     if ((CodecFlags&FFM_CODEC_FLAG_GLOBAL_HEADER)!=0)
         ctx->avctx->flags |= AV_CODEC_FLAG_GLOBAL_HEADER;
@@ -604,7 +605,7 @@ int __cdecl ffm_audio_get_codec_information(FFM_CodecInfo* info, const char* nam
     info->capabilities = codec->capabilities;
     if (codec->profiles) {
         for (i=0;i<32;i++) {
-            if (codec->profiles[i].profile==FF_PROFILE_UNKNOWN)
+            if (codec->profiles[i].profile==AV_PROFILE_UNKNOWN)
                 break;
             info->profiles_names[i]=codec->profiles[i].name;
             info->profiles_values[i]=codec->profiles[i].profile;
