diff --git a/src/assistant/qlitehtml/src/3rdparty/litehtml/src/el_font.cpp b/src/assistant/qlitehtml/src/3rdparty/litehtml/src/el_font.cpp
index e8e1f7e..bb5c7ba 100644
--- a/src/assistant/qlitehtml/src/3rdparty/litehtml/src/el_font.cpp
+++ b/src/assistant/qlitehtml/src/3rdparty/litehtml/src/el_font.cpp
@@ -1,6 +1,8 @@
 #include "el_font.h"
 #include "document.h"
 
+#include <cstdlib>
+
 
 litehtml::el_font::el_font(const std::shared_ptr<document>& doc) : html_tag(doc)
 {
