From 01c3bf079f49332d28d3ff4830fda1fc75ed077b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Trojnara?= <Michal.Trojnara@stunnel.org>
Date: Mon, 16 Mar 2026 16:34:15 +0100
Subject: [PATCH] Build fix for opaque ASN1_STRING

Updates #631
---
 src/p11_ec.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/p11_ec.c b/src/p11_ec.c
index 10c10bfc..4495e708 100644
--- a/src/p11_ec.c
+++ b/src/p11_ec.c
@@ -279,8 +279,8 @@ static int pkcs11_get_point(EC_KEY *ec, PKCS11_OBJECT_private *key, CK_SESSION_H
 	a = point;
 	os = d2i_ASN1_OCTET_STRING(NULL, &a, (long)point_len);
 	if (os) {
-		a = os->data;
-		rv = o2i_ECPublicKey(&ec, &a, os->length) == NULL;
+		a = ASN1_STRING_get0_data(os);
+		rv = o2i_ECPublicKey(&ec, &a, ASN1_STRING_length(os)) == NULL;
 		ASN1_STRING_free(os);
 	}
 	if (rv) { /* Workaround for broken PKCS#11 modules */
