diff --git a/src/lib/utils/mem_pool/mem_pool.cpp b/src/lib/utils/mem_pool/mem_pool.cpp
index 5e3caf2..0d3a534 100644
--- a/src/lib/utils/mem_pool/mem_pool.cpp
+++ b/src/lib/utils/mem_pool/mem_pool.cpp
@@ -9,6 +9,7 @@
 #include <botan/mem_ops.h>
 #include <botan/internal/target_info.h>
 #include <algorithm>
+#include <exception>
 #include <optional>
 
 #if defined(BOTAN_HAS_VALGRIND) || defined(BOTAN_ENABLE_DEBUG_ASSERTS)
