diff --git a/processcore/plugins/network/helper/Accumulator.cpp b/processcore/plugins/network/helper/Accumulator.cpp
index 2153086..3df1a14 100644
--- a/processcore/plugins/network/helper/Accumulator.cpp
+++ b/processcore/plugins/network/helper/Accumulator.cpp
@@ -9,6 +9,8 @@
 #include "Capture.h"
 #include "ConnectionMapping.h"
 
+#include <algorithm>
+
 using namespace std::chrono_literals;
 
 Accumulator::Accumulator(std::shared_ptr<Capture> capture, std::shared_ptr<ConnectionMapping> mapping)
