diff --git a/plasmoid/lib/syncthingapplet.cpp b/plasmoid/lib/syncthingapplet.cpp
index 6f2e6693..05da97b9 100644
--- a/plasmoid/lib/syncthingapplet.cpp
+++ b/plasmoid/lib/syncthingapplet.cpp
@@ -225,7 +225,7 @@ void SyncthingApplet::initEngine(QObject *object)
     if (!engine) {
         return;
     }
-    const auto color = m_theme.color(Plasma::Theme::TextColor, Plasma::Theme::NormalColorGroup);
+    const auto color = m_theme.color(Plasma::Theme::TextColor);
     m_imageProvider = new QtForkAwesome::QuickImageProvider(m_iconManager.forkAwesomeRenderer(), color);
     connect(engine, &QObject::destroyed, this, &SyncthingApplet::handleImageProviderDestroyed); // engine has ownership over image provider
     engine->addImageProvider(QStringLiteral("fa"), m_imageProvider);
@@ -720,7 +720,7 @@ void SyncthingApplet::handleThemeChanged()
         m_iconManager.setPalette(palette);
         setBrightColors(isPaletteDark(palette));
         if (m_imageProvider) {
-            m_imageProvider->setDefaultColor(m_theme.color(Plasma::Theme::TextColor, Plasma::Theme::NormalColorGroup));
+            m_imageProvider->setDefaultColor(m_theme.color(Plasma::Theme::TextColor));
         }
         emit faUrlChanged(m_faUrl = QStringLiteral("image://fa/"));
     });
