diff --git a/src/plugins/overview/qml/Main.qml b/src/plugins/overview/qml/Main.qml
index 584a362d5a..1b9b33ef74 100644
--- a/src/plugins/overview/qml/Main.qml
+++ b/src/plugins/overview/qml/Main.qml
@@ -343,7 +343,7 @@ FocusScope {
 
     property int desktops: Math.max(bar.desktopCount, 2)
     property int columns: Math.ceil(desktops / rows)
-    property int rows: KWinComponents.Workspace.desktopGridHeight
+    property int rows: Math.round(Math.sqrt(desktops))
 
     // The desktop might shuffle around as soon as it's
     // created since the rows/columns are updated after
