diff --git a/src/3rdparty/chromium/sandbox/linux/system_headers/linux_seccomp.h.orig b/src/3rdparty/chromium/sandbox/linux/system_headers/linux_seccomp.h
index 8690a96..8ebf404 100644
--- a/src/3rdparty/chromium/sandbox/linux/system_headers/linux_seccomp.h.orig
+++ b/src/3rdparty/chromium/sandbox/linux/system_headers/linux_seccomp.h
@@ -214,8 +214,4 @@ struct seccomp_notif_addfd {
 #define SECCOMP_RET_INVALID 0x00010000U  // Illegal return value
 #endif
 
-#ifndef SYS_SECCOMP
-#define SYS_SECCOMP                   1
-#endif
-
 #endif  // SANDBOX_LINUX_SYSTEM_HEADERS_LINUX_SECCOMP_H_
