diff --git a/src/3rdparty/chromium/components/autofill/core/browser/metrics/quality_metrics.cc b/src/3rdparty/chromium/components/autofill/core/browser/metrics/quality_metrics.cc
index c91a7c1..735af37 100644
--- a/src/3rdparty/chromium/components/autofill/core/browser/metrics/quality_metrics.cc
+++ b/src/3rdparty/chromium/components/autofill/core/browser/metrics/quality_metrics.cc
@@ -77,8 +77,9 @@ void LogDurationMetrics(const FormStructure& form,
                         base::TimeTicks interaction_time,
                         base::TimeTicks submission_time) {
   size_t num_detected_field_types =
-      std::ranges::count_if(form, &FieldHasMeaningfulPossibleFieldTypes,
-                            &std::unique_ptr<AutofillField>::operator*);
+      std::ranges::count_if(form, [](const auto& field) {
+        return FieldHasMeaningfulPossibleFieldTypes(*field);
+      });
   bool form_has_autofilled_fields = std::ranges::any_of(
       form, [](const auto& field) { return field->is_autofilled(); });
   bool has_observed_one_time_code_field =
diff --git a/src/3rdparty/chromium/content/browser/cookie_store/cookie_store_manager.cc b/src/3rdparty/chromium/content/browser/cookie_store/cookie_store_manager.cc
index 3b514fc..fd8e511 100644
--- a/src/3rdparty/chromium/content/browser/cookie_store/cookie_store_manager.cc
+++ b/src/3rdparty/chromium/content/browser/cookie_store/cookie_store_manager.cc
@@ -231,9 +231,10 @@ void CookieStoreManager::AddSubscriptions(
     auto new_subscription = std::make_unique<CookieChangeSubscription>(
         std::move(mojo_subscription), service_worker_registration->id());
 
-    auto existing_subscription_it = std::ranges::find(
-        subscriptions, *new_subscription,
-        &std::unique_ptr<CookieChangeSubscription>::operator*);
+    auto existing_subscription_it = std::ranges::find_if(
+        subscriptions, [&new_subscription](const auto& sub) {
+          return *sub == *new_subscription;
+        });
     if (existing_subscription_it == subscriptions.end())
       subscriptions.push_back(std::move(new_subscription));
   }
@@ -329,9 +330,10 @@ void CookieStoreManager::RemoveSubscriptions(
   }
 
   for (auto& subscription : all_subscriptions) {
-    auto target_subscription_it = std::ranges::find(
-        target_subscriptions, *subscription,
-        &std::unique_ptr<CookieChangeSubscription>::operator*);
+    auto target_subscription_it = std::ranges::find_if(
+        target_subscriptions,
+        [&subscription](const auto& sub) { return *sub == *subscription; });
+
     if (target_subscription_it == target_subscriptions.end()) {
       // The subscription is not marked for deletion.
       live_subscriptions.push_back(std::move(subscription));
