diff --git a/shared/source/host_function/host_function.h b/shared/source/host_function/host_function.h
index ae01467..e944366 100644
--- a/shared/source/host_function/host_function.h
+++ b/shared/source/host_function/host_function.h
@@ -15,6 +15,7 @@
 #include <mutex>
 #include <optional>
 #include <unordered_map>
+#include <vector>
 
 #if defined(_WIN32)
 #if !defined(NEO_HOST_FUNCTION_CALLBACK)
