From df789ec326ae0f2c619f91c8f2fc8b5e45b50a70 Mon Sep 17 00:00:00 2001
From: FrozenGalaxy <156858866+FrozenGalaxy@users.noreply.github.com>
Date: Sun, 18 May 2025 02:03:22 +0200
Subject: [PATCH] update moviedecoder.cpp for newest ffmpeg

---
 libffmpegthumbnailer/moviedecoder.cpp | 29 +++++++++++++++++----------
 1 file changed, 18 insertions(+), 11 deletions(-)

diff --git a/libffmpegthumbnailer/moviedecoder.cpp b/libffmpegthumbnailer/moviedecoder.cpp
index f19f082..68170df 100644
--- a/libffmpegthumbnailer/moviedecoder.cpp
+++ b/libffmpegthumbnailer/moviedecoder.cpp
@@ -406,7 +406,7 @@ void MovieDecoder::initializeFilterGraph(const AVRational& timeBase, const std::
             "Failed to create filter sink");
 
     AVFilterContext* yadifFilter = nullptr;
-    if (m_pFrame->interlaced_frame != 0)
+    if (m_pFrame->flags & AV_FRAME_FLAG_INTERLACED)
     {
         checkRc(avfilter_graph_create_filter(&yadifFilter, avfilter_get_by_name("yadif"), "thumb_deint", "deint=1", nullptr, m_pFilterGraph),
                 "Failed to create deinterlace filter");
@@ -518,7 +518,7 @@ void MovieDecoder::seek(int timeInSeconds)
         }
 
         ++keyFrameAttempts;
-    } while ((!gotFrame || !m_pFrame->key_frame) && keyFrameAttempts < 200);
+    } while ((!gotFrame || !(m_pFrame->flags & AV_FRAME_FLAG_KEY)) && keyFrameAttempts < 200); 
 
     if (gotFrame == 0)
     {
