diff --git a/tesseract/src/arch/simddetect.cpp.orig b/tesseract/src/arch/simddetect.cpp
index 81d7c07..b1a4566 100644
--- a/tesseract/src/arch/simddetect.cpp.orig
+++ b/tesseract/src/arch/simddetect.cpp
@@ -25,6 +25,8 @@
 #include "simddetect.h"
 #include "tprintf.h" // for tprintf
 
+#include <stdlib.h>
+
 #if !defined(__clang__) && defined(__GNUC__) && (__GNUC__ < 12)
 // The GNU compiler g++ fails to compile with the Accelerate framework
 // (tested with versions 10 and 11), so unconditionally disable it.
