From fff75d62198a443c85a05460a35a34ccfa76c07b Mon Sep 17 00:00:00 2001
From: Xarblu <xarblu@protonmail.com>
Date: Tue, 3 Mar 2026 23:00:08 +0100
Subject: [PATCH] fix build with clang-21+

Signed-off-by: Xarblu <xarblu@protonmail.com>
---
 bpf-internal.h | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/bpf-internal.h b/bpf-internal.h
index 38b79a9e1..fade42b23 100644
--- a/bpf-internal.h
+++ b/bpf-internal.h
@@ -224,17 +224,19 @@ const opcode BPF_LD_MAP = BPF_LD | BPF_IMM | BPF_DW | (BPF_PSEUDO_MAP_FD << 8);
   FN(text_str),                   \
   FN(string_quoted),
  
-const bpf_func_id BPF_FUNC_map_get_next_key    = (bpf_func_id) -1;
-const bpf_func_id BPF_FUNC_sprintf             = (bpf_func_id) -2;
-const bpf_func_id BPF_FUNC_stapbpf_stat_get    = (bpf_func_id) -3;
-const bpf_func_id BPF_FUNC_gettimeofday_ns     = (bpf_func_id) -4;
-const bpf_func_id BPF_FUNC_get_target          = (bpf_func_id) -5;
-const bpf_func_id BPF_FUNC_set_procfs_value    = (bpf_func_id) -6;
-const bpf_func_id BPF_FUNC_append_procfs_value = (bpf_func_id) -7;
-const bpf_func_id BPF_FUNC_get_procfs_value    = (bpf_func_id) -8;
-const bpf_func_id BPF_FUNC_str_concat          = (bpf_func_id) -9;
-const bpf_func_id BPF_FUNC_text_str            = (bpf_func_id) -10;
-const bpf_func_id BPF_FUNC_string_quoted       = (bpf_func_id) -11;
+enum {
+    BPF_FUNC_map_get_next_key    = (bpf_func_id) -1,
+    BPF_FUNC_sprintf             = (bpf_func_id) -2,
+    BPF_FUNC_stapbpf_stat_get    = (bpf_func_id) -3,
+    BPF_FUNC_gettimeofday_ns     = (bpf_func_id) -4,
+    BPF_FUNC_get_target          = (bpf_func_id) -5,
+    BPF_FUNC_set_procfs_value    = (bpf_func_id) -6,
+    BPF_FUNC_append_procfs_value = (bpf_func_id) -7,
+    BPF_FUNC_get_procfs_value    = (bpf_func_id) -8,
+    BPF_FUNC_str_concat          = (bpf_func_id) -9,
+    BPF_FUNC_text_str            = (bpf_func_id) -10,
+    BPF_FUNC_string_quoted       = (bpf_func_id) -11,
+};
 
 struct insn
 {
-- 
2.53.0

